src/layout.rs

```diff
@@ -265,7 +265,7 @@ impl Layout {
             .is_ok()
     }
 
-    /// Builder method to set the constraints of the layout.
+    /// Fluent setters to set the constraints of the layout.
```

Comment:
Sorry for my late reply. I prefer this suggestion. For a user, simple expressions can help read. We can fix the old wrong version and just use `set` for clearness.

> combine this with adding #[must_use] attributes on all the setters?

I'm not sure what to do about this. Do you mean check all setters and give the  `#[must_use]` & `set doc`?

Reply:
I think any setter method that takes self and returns self needs to have a #[must_use] attribute. This indicates that the caller must do something with the return value and makes it clear that it's an error instead of a bug to write code that doesn't use the return value like below. Because this changes the same code (i.e. every setter method in Ratatui), it's worth bundling this change into a single large change where we add the same language across the entire crate and also add that attribute so we don't miss any of the places where this is necessary. Alternatively just fix the comments here and we can split that out to two more tasks (add consistent comments for all setters and add must_use on all fluent setters).

```rust
let layout = Layout::default();
layout.constraints([Min(0)...]);
```

Reply:
I agree with "less is more". 

Let's not focus on the term for now. `not modifying the underlying object` is not the direct reason it must be used. 

When users invoke a setter, they tend to make some changes on the initial instance. I prefer to explain that:

 = note: `The setters have moved & returned the original instance, so the yield should be used or this operation is redundant.`

@joshka Do you think this wording fits your requirements? I think this is clear enough for new devs. `modifying` is the point I didn't get in the first place.

We can still express fluent setters in the code docs, I think it's ok to be formal there.

Reply:
That seems a bit wordy / repeats the existing not about using the value. What about:
"method moves the value of self and returns the modified value"?

Reply:
Great! Very straightforward. It's so nice to discuss with you. I'll follow this on my next commit.